eng/tools/compare-packages.ps1

```diff
@@ -0,0 +1,60 @@
+param (
+  $pathToMasterPkg,
+  $pathToCurrentPkg,
+  $dailyDevBuildNo,
+  $pathForDiffFile
+)
+
+function ExtractTGZPackages($pathToPkg){
+  $regExp = "-\d+\.\d+\.\d+(-\w*\.\d*)?(-dev-$dailyDevBuildNo)?"
+
+  $parentExtractDir = Join-Path $pathToPkg -ChildPath "all-contents"
+  Write-Host "mkdir $parentExtractDir"
+  mkdir $parentExtractDir
+
+  foreach ($p in $(dir $pathToPkg -r -i *.tgz)){
+    $extractDir = Join-Path $parentExtractDir -ChildPath $p.BaseName
+
+    if($p.BaseName -match $regExp){
+      $newDirName = ($p.BaseName -split $Matches.0)[0]
```

Comment:
I'm not sure what you are trying to do with this regex. Are you attempting to remove the version? If so I'd do it all the time and simply the regex to something like:
```
if ($p.BaseName -match "(?<name>.*)(-\d.*)") {
  $extractDir = Join-Path $parentExtractDir $matches["name"]
}
else {
   write-error "Package name $($p.BaseName) doesn't match the expected format [<name>-<version>]!"
}
```

Reply:
Yeah I am trying to match the version - GA format, preview format and Dev build version format. After that, I am removing the version from the name and making a directory with just that name. This is the directory where the respective package is extracted. So it will be easier to get a proper diff of the file content with master/azure-service-bus and PR/azure-service-bus rather than either of them having a version @weshaggard